feature_engine/encoding/one_hot.py

```diff
@@ -160,9 +175,12 @@ class OneHotEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
     def __init__(
         self,
         top_categories: Optional[int] = None,
+        custom_categories: Optional[Dict] = None,
         drop_last: bool = False,
         drop_last_binary: bool = False,
-        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        variables: Union[
+            None, int, str, List[Union[str, int]], Iterable[Union[str, int]]
```

Comment:
why did you add iterable?

Reply:
Before refactoring the code, MyPy raised the following errors:

```
feature_engine/encoding/one_hot.py:221: error: Argument 1 to "sorted" has incompatible type "Union[int, str, List[Union[str, int]], None]"; expected "Iterable[Union[str, int]]"
feature_engine/encoding/one_hot.py:223: error: Argument 1 to "sorted" has incompatible type "Union[int, str, List[Union[str, int]], None]"; expected "Iterable[Union[str, int]]"
```